integreat_cms/cms/forms/media/upload_media_file_form.py

```diff
@@ -124,19 +124,20 @@ def clean(self) -> dict:
                 ),
             )
 
-        # If everything looks good until now, generate a thumbnail and an optimized image
         if (
             not self.errors
             and (img_type := cleaned_data.get("type"))
             and img_type.startswith("image")
         ):
-            if optimized_image := generate_thumbnail(
-                file,
-                settings.MEDIA_OPTIMIZED_SIZE,
-                False,
-            ):
-                cleaned_data["file"] = optimized_image
-                cleaned_data["thumbnail"] = generate_thumbnail(file)
+            if thumbnail := generate_thumbnail(file):
```

Comment:
Since this method is already getting a little bit out of hand, could you please put this functionality into it's own method? :) If you want to go the extra mile, you could add another commit refactoring this method a little bit (s. scout rule)[https://deviq.com/principles/boy-scout-rule] :)

Reply:
I tried, but I'm not really satisfied as it does not feel very clean cut. I'm open for any suggestions!

Reply:
Giving this function another look, I don't feel this way anymore :) For me we can resolve this conversation! Thank you for the improvement :)

Reply:
Leaving it as it now is, or leaving it as it originally was, without outsourced `process_…()` methods?

Reply:
Leaving at it is now :)